pkg/admission/mutator/shoot.go

```diff
@@ -53,21 +94,190 @@ func (s *shootMutator) Mutate(_ context.Context, new, old client.Object) error {
 
-func (s *shootMutator) mutateShootUpdate(oldShoot, shoot *corev1beta1.Shoot) error {
+	var (
+		secretBinding    = &corev1beta1.SecretBinding{}
+		secretBindingKey = kutil.Key(shoot.Namespace, shoot.Spec.SecretBindingName)
+	)
+	if err := kutil.LookupObject(ctx, s.client, s.apiReader, secretBindingKey, secretBinding); err != nil {
+		return false, err
+	}
+
+	var (
+		secret    = &corev1.Secret{}
+		secretRef = secretBinding.SecretRef.Name
+		secretKey = kutil.Key(secretBinding.SecretRef.Namespace, secretRef)
+	)
+	if err := s.apiReader.Get(ctx, secretKey, secret); err != nil {
+		return false, err
+	}
+	accessKeyID, ok := secret.Data[alicloud.AccessKeyID]
+	if !ok {
+		return false, fmt.Errorf("missing %q field in secret %s", alicloud.AccessKeyID, secretRef)
+	}
+	accessKeySecret, ok := secret.Data[alicloud.AccessKeySecret]
+	if !ok {
+		return false, fmt.Errorf("missing %q field in secret %s", alicloud.AccessKeySecret, secretRef)
+	}
+	shootECSClient, err := s.alicloudClientFactory.NewECSClient(region, string(accessKeyID), string(accessKeySecret))
+	if err != nil {
+		return false, err
+	}
+	if exist, err := shootECSClient.CheckIfImageExists(ctx, imageId); err != nil {
+		return false, err
+	} else if exist {
+		if ownedByAliCloud, err := shootECSClient.CheckIfImageOwnedByAliCloud(imageId); err != nil {
+			return false, err
+		} else if ownedByAliCloud {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+func (s *shootMutator) getImageId(ctx context.Context, imageName string, imageVersion *string, imageRegion string, cloudProfileName string) (string, error) {
+	var (
+		cloudProfile    = &corev1beta1.CloudProfile{}
+		cloudProfileKey = kutil.Key(cloudProfileName)
+	)
+	if err := kutil.LookupObject(ctx, s.client, s.apiReader, cloudProfileKey, cloudProfile); err != nil {
+		return "", err
+	}
+	cloudProfileConfig, err := s.getCloudProfileConfig(cloudProfile)
+	if err != nil {
+		return "", err
+	}
+	return helper.FindImageForRegionFromCloudProfile(cloudProfileConfig, imageName, *imageVersion, imageRegion)
+}
+
+func (s *shootMutator) getCloudProfileConfig(cloudProfile *corev1beta1.CloudProfile) (*api.CloudProfileConfig, error) {
+	var cloudProfileConfig *api.CloudProfileConfig = &api.CloudProfileConfig{}
+	if _, _, err := s.decoder.Decode(cloudProfile.Spec.ProviderConfig.Raw, nil, cloudProfileConfig); err != nil {
+		return nil, fmt.Errorf("could not decode providerConfig of cloudProfile for '%s': %w", kutil.ObjectName(cloudProfile), err)
+	}
+
+	return cloudProfileConfig, nil
+}
+func (s *shootMutator) mutateShootUpdate(ctx context.Context, oldShoot, shoot *corev1beta1.Shoot) error {
+	if !equality.Semantic.DeepEqual(oldShoot.Spec, shoot.Spec) {
+		if err := s.mutateShootUpdateForEncryptedDisk(ctx, oldShoot, shoot); err != nil {
+			return err
+		}
+	}
 	if !equality.Semantic.DeepEqual(oldShoot.Spec, shoot.Spec) {
 		s.mutateForEncryptedSystemDiskChange(oldShoot, shoot)
```

Comment:
Please help rename the function name as `triggerInfraUpdateForNewEncryptedSystemDisk`.  `mutateForEncryptedSystemDiskChange` and `mutateShootUpdateForEncryptedDisk` like twins....